yjit/src/core.rs

```diff
@@ -411,43 +411,101 @@ impl From<Opnd> for YARVOpnd {
     }
 }
 
+/// Number of registers that can be used for stack temps
+pub const MAX_TEMP_REGS: usize = 5;
+
 /// Maximum index of stack temps that could be in a register
 pub const MAX_REG_TEMPS: u8 = 8;
 
-/// Bitmap of which stack temps are in a register
-#[derive(Copy, Clone, Default, Eq, Hash, PartialEq, Debug)]
-pub struct RegTemps(u8);
+/// A stack slot or a local variable. u8 represents the index of it (<= 8).
+#[derive(Copy, Clone, Eq, Hash, PartialEq, Debug)]
+pub enum RegTemp {
+    Stack(u8),
+    Local(u8),
+}
```

Comment:
I find the name `RegTemp` confusing here.

Usually temp means a temporary (like a value on the stack), so a local is not a temp.

Does `RegTemp` represent something a register can be mapped to? If so, can we rename it to `RegMapping` ?

Reply:
Or looking at the code, I guess `RegTemp` is meant to represent an operand that is register-allocatable?

Reply:
> Does RegTemp represent something a register can be mapped to? If so, can we rename it to RegMapping ?
>
> Or looking at the code, I guess RegTemp is meant to represent an operand that is register-allocatable?

I'd say `RegTemps` are a mapping as a whole (the former) but a `RegTemp` is an operand (the latter). I just renamed both of them to `RegMapping`(s) https://github.com/ruby/ruby/pull/11157/commits/a7155c2b0bdaefd87b5ebc0a7802f427ee7296ec, but maybe `RegMapping` (former `RegTemp`) could be `RegOperand`? 🤔

Reply:
Maybe `RegOpnd` for consistency with other `Opnd` in the codebase 🤔 ?

Reply:
:+1: https://github.com/ruby/ruby/pull/11157/commits/35119a841462aede6748565bbb2d33469f20cccb